fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/event/player/PlayerBlockPlaceEvents.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.player;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+public final class PlayerBlockPlaceEvents {
+	private PlayerBlockPlaceEvents() { }
+
+	/**
+	 * Callback before a block is placed.
+	 * Called on server only
+	 *
+	 * <p>If any listener cancels a block placing action, that block placing
+	 * action is canceled and {@link #CANCELED} event is fired. Otherwise, the
+	 * {@link #AFTER} event is fired.</p>
+	 */
+	public static final Event<Before> BEFORE = EventFactory.createArrayBacked(Before.class,
```

Comment:
Upon further discussion about the block break events, Player and I agree that the cancelation and before phases of the event need to be separated.

This would be Split into and `Allow` and `Before`, where the before is no longer Cancellable.
Allow would be dispatched first and then fire the before and after or canceled depending on the result of the Allow phase.